src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvPhoneServiceChannelEntityUpdater.java

```diff
@@ -0,0 +1,143 @@
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import fi.metatavu.restfulptv.client.ApiResponse;
+import fi.metatavu.restfulptv.client.model.PhoneServiceChannel;
+import fi.otavanopisto.kuntaapi.server.cache.ModificationHashCache;
+import fi.otavanopisto.kuntaapi.server.controllers.IdentifierController;
+import fi.otavanopisto.kuntaapi.server.discover.EntityUpdater;
+import fi.otavanopisto.kuntaapi.server.id.IdController;
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.id.PhoneServiceChannelId;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiIdFactory;
+import fi.otavanopisto.kuntaapi.server.integrations.ptv.resources.PtvPhoneServiceChannelResourceContainer;
+import fi.otavanopisto.kuntaapi.server.integrations.ptv.tasks.PhoneServiceChannelIdTaskQueue;
+import fi.otavanopisto.kuntaapi.server.persistence.model.Identifier;
+import fi.otavanopisto.kuntaapi.server.settings.SystemSettingController;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask.Operation;
+
+@ApplicationScoped
+@Singleton
+@AccessTimeout (unit = TimeUnit.HOURS, value = 1l)
+@SuppressWarnings ("squid:S3306")
+public class PtvPhoneServiceChannelEntityUpdater extends EntityUpdater {
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private SystemSettingController systemSettingController;
+
+  @Inject
+  private PtvApi ptvApi;
+
+  @Inject
+  private KuntaApiIdFactory kuntaApiIdFactory;
+  
+  @Inject
+  private PtvTranslator ptvTranslator;
+
+  @Inject
+  private PtvIdFactory ptvIdFactory;
+
+  @Inject
+  private IdController idController;
+
+  @Inject
+  private IdentifierController identifierController;
+
+  @Inject
+  private PtvPhoneServiceChannelResourceContainer ptvPhoneServiceChannelResourceContainer;
+
+  @Inject
+  private ModificationHashCache modificationHashCache;
+  
+  @Inject
+  private PhoneServiceChannelIdTaskQueue phoneServiceChannelIdTaskQueue;
+
+  @Resource
+  private TimerService timerService;
+
+  @Override
+  public String getName() {
+    return "ptv-phone-channels";
+  }
+
+  @Override
+  public void timeout() {
+    executeNextTask();
+  }
+  
+  @Override
+  public TimerService getTimerService() {
+    return timerService;
+  }
+  
+  private void executeNextTask() {
+    IdTask<PhoneServiceChannelId> task = phoneServiceChannelIdTaskQueue.next();
+    if (task != null) {
+      PhoneServiceChannelId phoneServiceChannelId = task.getId();
+      
+      if (task.getOperation() == Operation.UPDATE) {
+        updateServiceChannelChannel(phoneServiceChannelId, task.getOrderIndex());
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteServiceChannelChannel(phoneServiceChannelId);
+      }
+    }
+  }
+
+  private void updateServiceChannelChannel(PhoneServiceChannelId ptvPhoneServiceChannelId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateServiceChannelChannel" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)